### PR TITLE
[lldb-dotest] Don't swallow error exit codes.

### PR DESCRIPTION
**Pull request by dcci**
<rdar://problem/39014459>

git-svn-id: https://llvm.org/svn/llvm-project/lldb/trunk@328894 91177308-0d34-0410-b5e6-96231b3b80d8

### CODE REVIEW

**Comment by dcci** [comment]
<rdar://problem/39014459>

**Comment by dcci** [comment]
@swift-ci test and merge

File: test/lldb-dotest.in [hunk 1]
```diff
@@ -20,4 +20,6 @@ if __name__ == '__main__':
         cmd.extend(['--swift-library', swift_library])
     cmd.extend(wrapper_args)
     # Invoke dotest.py
-    subprocess.call(cmd)
+    exit_code = subprocess.call(cmd)
+    if exit_code != 0:
```
Comment (keith) [inline test/lldb-dotest.in:24]: no need to have the if, you could `sys.exit(subprocess.call(cmd))`

**Comment by dcci** [comment]
Hey @Keith, thanks. You can tell from my code I'm definitely not a pythonista :)
This is backported from llvm.org so I might consider making the change there and back port the new version. Thanks for your comment. 

[FWIW, I'm rushing this in because the CMake bot on `ci.swift.org` does not report a failure if some of the tests are failing, but your cleanup looks really good so I'm going to do it as a follow-up]

**Comment by keith** [comment]
No big deal either way! Just thought I'd mention it 😄

**Comment by JDevlieghere** [comment]
Thanks @dcci!

Committed @keith's suggestion upstream in `r328966`, I'll include it in my next cherry-pick. Thanks!

**Comment by dcci** [comment]
Thank you Jonas.

**Pull request merged**